### ISSUE TITLE
Cocoon Connect RGB LED Strip - Not working

### ISSUE THREAD

**Issue by TheHomekitWayFinder** [iRayanKhan/homebridge-tuya-platform#15]
I have just purchased a new [Cocoon Connect LED Strip](https://cocoonproducts.com.au/cocoonconnect/smart-led-light-strip/) from Aldi and have successfully acquired the I’d and key of the device and set the device up as a `SimpleLight` and `RGBTWLIGHT` but neither of them are working. I have got the schema and signature of the device to help with troubleshooting. The LED strip doesn’t appear to support colour temperature but I’m not sure. Any help would be appreciated.

Schema:

` {
    name: 'LED Strip',
    id: 'xxxx',
    key: 'xxxx',
    def: {
      schema: '[{"mode":"rw","code":"switch_led","name":"\\u5f00\\u5173","property":{"type":"bool"},"iconname":"icon-dp_power2","id":20,"type":"obj","desc":""},{"mode":"rw","code":"work_mode","name":"\\u6a21\\u5f0f","property":{"range":["white","colour","scene","music"],"type":"enum"},"iconname":"icon-dp_list","id":21,"type":"obj","desc":""},{"mode":"rw","code":"colour_data","name":"\\u5f69\\u5149","property":{"type":"string","maxlen":255},"iconname":"icon-dp_light","id":24,"type":"obj","desc":""},{"mode":"rw","code":"scene_data","name":"\\u573a\\u666f","property":{"type":"string","maxlen":255},"iconname":"icon-dp_box2","id":25,"type":"obj","desc":""},{"mode":"rw","code":"countdown","name":"\\u5012\\u8ba1\\u65f6\\u5269\\u4f59\\u65f6\\u95f4","property":{"unit":"","min":0,"max":86400,"scale":0,"step":1,"type":"value"},"iconname":"icon-timer1","id":26,"type":"obj","desc":""},{"mode":"wr","code":"control_data","name":"\\u8c03\\u8282","property":{"type":"string","maxlen":255},"iconname":"icon-setting","id":28,"type":"obj","desc":""}]',
      extras: '[{"id":20,"inputStyle":"","inputType":" "},{"id":21,"inputStyle":"","inputType":" "},{"id":26,"inputStyle":"","inputType":" "}]',
    },
  },
]`

Signature:

`[1/13/2020, 2:48:13 PM] [TuyaLan] Ready to handle LED Strip (RGBTWLight:3.3) with signature {"20":true,"21":"colour","24":"004303e80139","25":"000e0d00002e03e802cc00000000","26":0}`

I believe that “20” is power state, “21” is mode (not sure though), “24” is colour and brightness, not sure what “25” is or “26” and neither of these change.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by iRayanKhan** [comment]
I'll try to add it, a bit behind on issues, but I will definitely get it added.

**Comment by TheHomekitWayFinder** [comment]
@iRayanKhan have you started work on this or not? If you need anything just ask

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by iRayanKhan** [comment]
I am currently, I am working my way trough issues at the moment so hope to get this done in the next few days. Contact sensors are really a pain xD

**Comment by TheHomekitWayFinder** [comment]
Yeah, it’s due to the nature of them not being always connected and only connecting when they are triggered. I had one but I took it back as it was going to be a nightmare with HomeKit and instead I’m going to get aqara sensors as I already have the aqara hub 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by iRayanKhan** [comment]
Can you paste your config I got it working out.

**Comment by TheHomekitWayFinder** [comment]
For the LED strip 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by iRayanKhan** [comment]
Yes, I don't know if you meant to attach it, but it didn't attach lol.

**Comment by TheHomekitWayFinder** [comment]
Device Config:
`{
                    "name": "LED Strip",
                    "type": "RGBTWLight",
                    "manufacturer": "Cocoon",
                    "model": "LED Strip",
                    "id": "xxxxxxxxxx",
                    "key": "xxxxxxxxxxx”
                } `

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by iRayanKhan** [comment]
If the schema analysis is correct make this your config:
```json
{
					"name": "LED Strip",
					"type": "RGBTWLight",
					"manufacturer": "Cocoon",
					"model": "LED Strip",
					"id": "xxxxxxxxxx",
					"key": "xxxxxxxxxxx",
					"dpPower": 20,
					"dpBrightness": 24,
					"dpColor": 24
	}
```
The other modes you described the only possible states for them could be:

 "dpPower": 1,
    "dpMode": 2,
    "dpBrightness": 3,

 
    "dpColorTemperature": 4,

    "dpColor": 5,
